### PR TITLE
chore: Fix control response

### PR DESCRIPTION
**Pull request by enikonovad**
Signed-off-by: Ekaterina Nikonova <enikonovad@gmail.com>

## Types of changes

- [ ] Hot fix (emergency fix and release)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [ ] Documentation for my change is up to date?
- [ ] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary
Correct control response method

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: tests/data/author/ssp/sample_nist_catalog.json [hunk 1]
```diff
@@ -0,0 +1,667 @@
+{
```
Comment (butler54) [inline tests/data/author/ssp/sample_nist_catalog.json:1]: The only comment I have here is can some of the existing catalogs / ssp's be leveraged? I'm reticent to add yet more test files which introduce the potential refactoring effort on a new release.

File: trestle/core/catalog_interface.py [hunk 2]
```diff
@@ -238,6 +238,46 @@ def get_all_groups_from_catalog(self) -> Iterator[cat.Group]:
                 for res in CatalogInterface._get_groups_from_group(my_group):
                     yield res
 
+    def get_statement_label_if_exists(self, control_id: str, statement_id: str) -> Optional[str]:
+        """Get statement label if given."""
+        # TODO: workaround when statement ids contain additional dot at the end
+        # remove this after this is fixed
+        if statement_id[-1] == '.':
+            statement_id = statement_id[0:-1]
```
Comment (butler54) [inline trestle/core/catalog_interface.py:246]: Let's see if we can merge @fsuits' PR first so we can remove this.

File: trestle/core/ssp_io.py [hunk 3]
```diff
@@ -213,14 +215,53 @@ def get_control_response(self, control_id: str, level: int) -> str:
...
+            for statement in control_impl_req.statements:
+                statement_id = statement.statement_id
+                label = statement_id
+                # look up label for this statement
+                if control.parts:
+                    found_label = self._catalog_interface.get_statement_label_if_exists(control_id, statement_id)
+                    if found_label:
+                        label = found_label
+
+                header = f'Implementation {label}'
```
Comment (butler54) [inline trestle/core/ssp_io.py:234]: So I think there are some extra checks needed here. If there is only one implemented requirement and no 'parts'  
then it should not have an implementation header. This is analgous to if the statement id corresponds to the top level statement for a control. 

If there is more than one part (and / or at multiple levels) - then you print the implementation ID.

File: trestle/core/ssp_io.py [hunk 4]
```diff
@@ -213,14 +215,53 @@ def get_control_response(self, control_id: str, level: int) -> str:
...
+                    for component_key in response_per_component:
+                        md_writer.new_header(level=level + 1, title=component_key)
+                        md_writer.set_indent_level(-1)
+                        md_writer.new_line(response_per_component[component_key])
+                        md_writer.set_indent_level(-1)
+
+        lines = md_writer.get_lines()
+
+        tree = MarkdownNode.build_tree_from_markdown(lines)
         tree.change_header_level_by(level)
```
Comment (butler54) [inline trestle/core/ssp_io.py:265]: Should this be a change by level command or do we need a 'set to the top level in this block is X' command?

File: trestle/core/ssp_io.py [hunk 5]
```diff
@@ -213,14 +215,53 @@ def get_control_response(self, control_id: str, level: int) -> str:
...
+                        # look up component title
+                        subheader = component.uuid
+                        response = ''
+                        if self._ssp.system_implementation.components:
+                            for comp in self._ssp.system_implementation.components:
+                                title = comp.title
+                                if title == SSP_MAIN_COMP_NAME and idx == 0:
+                                    # special case dont print first "This system"
+                                    continue
+                                subheader = title
```
Comment (butler54) [inline trestle/core/ssp_io.py:247]: Something looks strange here. It looks like it missing another comparison operation. (e.g. if component.uuid == comp.uuid) do X.

**Review by butler54 — CHANGES_REQUESTED** [review]
Please look at the changes suggested. Something looks a bit strange to me still.

Also have a talk with @fsuits on re-use of common SSPs etc to see if we can avoid adding more test data unnecessarily.

File: tests/data/author/ssp/sample_nist_catalog.json (hunk 1 above)
Comment (enikonovad) [reply tests/data/author/ssp/sample_nist_catalog.json]: Removed extra data and used existing

File: trestle/core/catalog_interface.py (hunk 2 above)
Comment (enikonovad) [reply trestle/core/catalog_interface.py:247]: Will remove after Frank's PR is merged

File: trestle/core/ssp_io.py (hunk 3 above)
Comment (enikonovad) [reply trestle/core/ssp_io.py]: Thanks, fixed

File: trestle/core/ssp_io.py (hunk 4 above)
Comment (enikonovad) [reply trestle/core/ssp_io.py:274]: `MarkdownNode` will adjust the markdown headers by the given level. By default all generated from SSP markdown text will start from `#` level

File: trestle/core/ssp_io.py (hunk 5 above)
Comment (enikonovad) [reply trestle/core/ssp_io.py]: Good catch, fixed

File: trestle/core/ssp_io.py (hunk 3 above)
Comment (enikonovad) [reply trestle/core/ssp_io.py]: Now the `Part {label}` will only be printed if the part has the name `item`. If part name is `statement` then only `description` per component in `by-component` will be printed if any

**Review by butler54 — APPROVED** [review]

**Pull request merged**